pgxn/neon/file_cache.c

```diff
@@ -640,6 +640,7 @@ lfc_write(NRelFileInfo rinfo, ForkNumber forkNum, BlockNumber blkno,
 	rc = pwrite(lfc_desc, buffer, BLCKSZ, ((off_t)entry->offset*BLOCKS_PER_CHUNK + chunk_offs)*BLCKSZ);
 	if (rc != BLCKSZ)
 	{
+		LWLockRelease(lfc_lock);
```

Comment:
I don't know anything about this code, but this looked suspicious to me: we are performing `lfc_disable` after releasing the lock. Is that correct? If so I'll approve

Reply:
lfc_disable applies its own exclusive lock. In all other places except that lfc_didsable was called without lock hold.

Reply:
Sounds good